js/views/modals/Settings/advanced/ReloadTransactions.js

```diff
@@ -0,0 +1,60 @@
+import resyncBlockchain, {
+  isResyncAvailable,
+  isResyncingBlockchain,
+  events as resyncEvents,
+} from '../../../../utils/resyncBlockchain';
+import loadTemplate from '../../../../utils/loadTemplate';
+import BaseVw from '../../../baseVw';
+
+export default class extends BaseVw {
+  constructor(options = {}) {
+    const opts = {
+      initialState: {
+        isSyncing: isResyncingBlockchain(),
+        syncComplete: false,
+        isResyncAvailable: isResyncAvailable(),
+        ...options.initialState || {},
+      },
+      ...options,
+    };
+
+    super(opts);
+    this.listenTo(resyncEvents, 'resyncing',
+      () => this.setState({
+        isSyncing: true,
+        syncComplete: false,
+      }));
+    this.listenTo(resyncEvents, 'resyncComplete',
+      () => this.setState({ isSyncing: false }));
```

Comment:
This needs syncComplete: true, or the button in settings will never show the complete state.